rust/src/io/reader.rs

```diff
@@ -754,6 +823,66 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn read_with_delete() {
+        let arrow_schema = ArrowSchema::new(vec![
+            ArrowField::new("i", DataType::Int64, true),
+            ArrowField::new("f", DataType::Float32, false),
+        ]);
+        let schema = Schema::try_from(&arrow_schema).unwrap();
+
+        let store = ObjectStore::memory();
+        let path = Path::from("/foo");
+
+        // Write 10 batches.
+        let mut file_writer = FileWriter::try_new(&store, &path, schema.clone())
+            .await
+            .unwrap();
+        for batch_id in 0..10 {
+            let value_range = batch_id * 10..batch_id * 10 + 10;
+            let columns: Vec<ArrayRef> = vec![
+                Arc::new(Int64Array::from_iter(
+                    value_range.clone().collect::<Vec<_>>(),
+                )),
+                Arc::new(Float32Array::from_iter(
+                    value_range.map(|n| n as f32).collect::<Vec<_>>(),
+                )),
+            ];
+            let batch = RecordBatch::try_new(Arc::new(arrow_schema.clone()), columns).unwrap();
+            file_writer.write(&[batch]).await.unwrap();
+        }
+        file_writer.finish().await.unwrap();
+
+        let fragment = 123;
+
+        // delete even rows
+        let dv = DeletionVector::Bitmap(RoaringBitmap::from_iter((0..100).filter(|x| x % 2 == 0)));
+        let deletion_file = write_deletion_file(fragment, 0, &dv, &store).await.unwrap();
+
+        let mut frag_struct = Fragment::new(fragment);
+        frag_struct.deletion_file = deletion_file;
+
+        let manifest = Manifest::new(&schema, Arc::new(vec![frag_struct]));
+
+        let mut reader =
+            FileReader::try_new_with_fragment(&store, &path, fragment, Some(&manifest))
+                .await
+                .unwrap();
+        reader.with_row_id(true);
```

Comment:
If we don't ask for the row id, but there is a deletion vector, we don't get the row id back, right? Should we test for that?